src/main/cli/cli.c

```diff
@@ -4400,6 +4437,14 @@ STATIC_UNIT_TESTED void cliSet(const char *cmdName, char *cmdline)
                         cliSetVar(val, value);
                         valueChanged = true;
                     }
+                } else if ((val->type & VALUE_TYPE_MASK) == VAR_INT32) {
+                    int32_t value = strtol(eqptr, NULL, 10);
+
+                    // INT32s are limited to being symmetric, so we test both bounds with the same magnitude
+                    if (value <= val->config.d32Max && value >= -val->config.d32Max) {
```

Comment:
```suggestion
                    const uint32_t absValue = ABS(value);
                    if (absValue <= val->config.u32Max) {
```
twos complement signed value are nasty. Minimum negative value is -(2^31), but maximum is 2^31-1. So value >= -val->config.d32Max would refuse -(2^31) (it is impossible to set limit correctly).

Reply:
This is comparing to the max allowed value for this particular CLI item as set in settings.c:1054 (900000000 and 1800000000 in this case). It has nothing to do with INT32_MAX or 2^31. So this conditional is essentially:
`if (value <= 1800000000  && value >= -1800000000 ) {`